### DISCUSSION TITLE
Sorting by letter capitalisation

### DISCUSSION THREAD

**Discussion by catvatar** [SebastianMC/obsidian-custom-sort#114] [Q&A]
## I use capital letters to distinguish between self contained notes and aggregating notes 
So I would like notes starting with capital letters to be sorted first.  
### Example 
- note A
- Note A
- note B
- Note B

Sorted:
- Note A
- Note B
- note A
- note B

For now I don't see a clear way of doing this. My `sortspec.md` file looks like this
```
---
sorting-spec: |-
  target-folder: /*
    /folders
    < a-z
    /:files
    < a-z
    README
    sortspec
---
```

**Comment by SebastianMC** [comment]
Hi @catvatar 

the requested feature was a trivial extension to existing mechanisms and thus I added it in the [2.1.1 release](https://github.com/SebastianMC/obsidian-custom-sort/releases/tag/2.1.1) of the plugin under #115.

The sorting specification to be used (in connection with the 2.1.1 release) looks like:

```yaml
---
sorting-spec: |-
  target-folder: /*
  /folders
    < a-z
  /:files \C...
    < a-z
  /:files \l...
    < a-z
  /:files
    < a-z   
  /! README
  /! sortspec
---
```

Because it is not trivial, let me take the opportunity and explain the key details:

<img src='https://github.com/SebastianMC/obsidian-custom-sort/assets/23032356/49b2738f-407e-4fa6-b5fa-329d33c9ef3b' width=695></img>

### ad 1.
`/:files \C...` uses the regexp symbol `\C` in  connection with the wildcard `...` to match & catch all files which name starts with a capital letter (the `\C` represents any capital letter, `...` matches any sequence of characters). Items in this group are sorted alphabetically in ascending order(`< a-z`)

### ad 2.
`/:files \l...` uses the regexp symbol `\l` in  connection with the wildcard `...` to match & catch all files which name starts with a lowercase letter (the `\l` represents any lowercase letter, `...` matches any sequence of characters). Items in this group are sorted alphabetically in ascending order(`< a-z`)

### ad 3.
`/:files` introduced to catch all files not matching any other criteria. In this specific case these are files which name doesn't start with capital or lowercase character. This could be filename starting with number or by a letter which has no uppercase/lowercase variants. Items in this group are sorted alphabetically in ascending order(`< a-z`)

### ad 4.
`/! README` is tricky :-). The prefix `/!` gives this matching rule a priority higher than these from 1, 2 and 3. Then `README` obviously specifies file or folder with that exact name. The prefix `/!` is required, otherwise the file named 'README' would be matched by the rule 1 ('README' starts with a capital letter)

### ad 5.
`/! sortspec` is tricky too, simlarly to 4. The prefix `/!` gives this matching rule a priority higher than these from 1, 2 and 3. Then `sortspec` obviously specifies file or folder with that exact name. The prefix `/!` is required, otherwise the file named 'sortspec' would be matched by the rule 2 ('sortspec' starts with a lowercase letter)

I hope it brings some more clarity!

**Comment by SebastianMC** [comment]
Hi @catvatar 

did you have a chance to test the functionality in 2.1.1? Does it work for you?

**Reply by catvatar** [reply]
Hi @SebastianMC 
Thank you for your detailed answer. 
Yes, after updating your plugin and adding changes like you suggested it works exactly as I wanted it to work.
